File: 001.TwoSum/main.cpp
#include "solution.h"

int main() {
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
//    vector<int> nums = {3, 3};
//    int target = 6;
    auto solution = Solution();
    vector<int> result = solution.twoSum(nums, target);
    for (int x:result) {
        cout << x << endl;
    }
    return 0;
}

File: 001.TwoSum/main.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
	let exist;
	const len = nums.length;

	for (let index = 0; index < len; index++) {
		const has = nums.indexOf(target - nums[index]);
		if (has !== -1 && has !== index) {
			exist = [index, has];
			break;
		}
	}

	return exist;
};

const nums = [3, 2, 4], target = 6;
const n = twoSum(nums, target);
console.log(n);